knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,57 @@
+# Title
+
+## Status
+
+proposed
+
+## Context 1
+
+When generating only from one spec (either generator), which is probably the most common use case, when doing this mannually, the resulting folder structure is `<outputDir>/serviceDir`, which in turn is unexpected.
+The name of the generator command is: `generate-X-client` indicating that only one client will be generated.
+
+## Decision 1
+
+1. Different behavior if only one specification is found vs. multiple.
+   For one specification, put client directly into <outputDir>, for multiple make subdirectories.
+2. generate-openapi-client batch -i <inputFile> -o <outputDir>
+   Generate only the first matched input file found by default.
+   Show a log message in case there are multiple inputs found, that batch argument is required to nest the APIs.
+3. generate-openapi-client -i <inputFile> -o <outputDir> --flat
+   Flatten the directory to be <outputDir>.
+   Throw an error if input is a directory (with multiple files).
+4. Keep it as is.
```

Comment:
I would keep is at least before 2.0, because:
- no breaking changes
- no customer requests
- I would be confused if generating two clients has different folder structure V.S. generating one client
May be we can rename to `generate-X-clients`

Reply:
Decision: Keep it as is.